src/App.tsx

```diff
@@ -15,48 +18,118 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SORT_FIELD_NAME: SortType = SortType.Name;
+const SORT_FIELD_LENGTH: SortType = SortType.Length;
+
+type Goods = string[];
+
+function getPreparedGoods(
+  goods: Goods,
+  { sortField, isReversed }: SortingParams,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_NAME:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.None);
+  const [isReversed, setReversed] = useState<boolean>(false);
+  const [isResetVisible, setIsResetVisible] = useState<boolean>(false);
```

Comment:
This is an overkill, this value should be calculated from `sortField` and `isReversed`, but should not be saved to state, because this will trigger extra renders